wpilib/src/dio.rs

```diff
@@ -154,6 +120,42 @@ impl OutputPin for DigitalOutput {
     }
 }
 
+/// Digital PWM output.
+///
+/// PWM output will be disabled when this is dropped.
+///
+/// Use [`DigitalOutput::enable_pwm`] to get an instance of this.
+///
+/// [`DigitalOutput::enable_pwm`]: struct.DigitalOutput.html#method.enable_pwm
+pub struct DigitalPwm<'a> {
+    _out: &'a mut DigitalOutput,
+    pwm: HAL_DigitalPWMHandle,
+}
+
+impl DigitalPwm<'_> {
+    /// Set a new duty cycle to use in PWM on this output.
+    pub fn update_duty_cycle(&mut self, duty_cycle: f64) -> HalResult<()> {
+        hal_call!(HAL_SetDigitalPWMDutyCycle(self.pwm, duty_cycle))
+    }
+
+    /// Set the PWM rate for digital PWM outputs, from 0.6Hz to 19kHz.
+    /// All digital channels will use the same PWM rate.
+    /// Will return an error if PWM has not been enabled.
```

Comment:
With the handle, isn't it always enabled?

Reply:
Good point. I probably forgot to remove this when moving things around.

Reply:
Actually re-reading this, this doesn't take a handle, so this comment is correct.

Reply:
Isn't this method only usable on a `DigitalPwm`, which will alwys have PWM enabled?

Reply:
This is an associated function, not a method.